src/server/chat/discord-manager.js

```diff
@@ -1,47 +1,121 @@
 
 const discordManager = function() {
+	let _discordClient = null;
+	let _chatWebhook = null; // Discord chat webhook
+	let _chatClientSocket = null;
+
+	let _onClientChatConnected = function(ws, req) {
+		// Get user if there is one
+		// Note: this might be pretty unsafe code. Remove or improve.
+		let name = "[Guest]";
+		let cookie = req.getHeader("cookie");
+		if (cookie) {
+			let cookies = cookie.split("; ");
+			let user_data = cookies.find(element => { return element.startsWith("user_data"); });
+			if (user_data) {
+				try {
+					user_data = decodeURIComponent(user_data);
+					user_data = user_data.substr(10);
+					user_data = JSON.parse(user_data);
+					if (db.user.idIsAuthenticated(user_data.id, user_data.access_token)) {
+						name = (`(${db.user.getUsernameById(user_data.id)})`).yellow;
+					} else {
+						name = "Hacker?!? (Authentication failed)".red;
+					}
+				}
+				catch (error) {
+					name = "Hacker?!? (Invalid cookie)".red;
+				}
+			}
+		}
+
+		log.info("A user connected! ".green + name);
+		ws.meta = { name };
+	};
+
+	let _onClientChatDisconnected = function(ws) {
+		log.info("A user disconnected... ".red + ws.meta.name);
+	};
+
+	let _onClientChatMessage = function(ws, message) {
+		try {
+			message = JSON.parse(message);
+		}
+		catch (e) {
+			ws.send("Invalid JSON");
+			return;
+		}
+
+		if (db.user.idIsAuthenticated(message.id, message.access_token)) {
+			let row = db.user.getUserDetailsById(message.id);
+			if (row) {
+				commandsManager.parse(message.content, message.id, row.username);
+
+				_chatWebhook.send(message.content, {
+					username: row.username,
+					avatarURL: `https://cdn.discordapp.com/avatars/${message.id}/${row.avatar}.png`,
+					disableEveryone: true
+				});
+
+				_chatClientSocket.emit(JSON.stringify({
+					username: row.username,
+					discriminator: row.discriminator,
+					content: message.content
+				}));
+			} else {
+				log.warn("User ID and access token mismatch!", row);
+			}
+		}
+	};
+
 	return {
-		initialize: function(db) {
-			const self = this;
-			// Set up clients
-			this.db = db;
-			this.client = new discord.Client();
-			this.chatWebhook = new discord.WebhookClient(config.discord.webhookId, config.discord.webhookToken);
+		initialize: function() {
+			_discordClient = new discord.Client();
+			_chatWebhook = new discord.WebhookClient(config.discord.webhookId, config.discord.webhookToken);
 
 			// Set up chat socket
-			const sockets = require("../network/sockets");
-			const socketChat = sockets.setupChat(db, this.chatWebhook);
+			_chatClientSocket = new socketsHelper.Socket("/chat", {
+				compression: 1,
+				maxPayloadLength: 128 * 1024,
+				idleTimeout: 3600
+			});
 
-			this.client.on("ready", () => {
+			_chatClientSocket.eventEmitter.on("open", _onClientChatConnected);
+			_chatClientSocket.eventEmitter.on("close", _onClientChatDisconnected);
+			_chatClientSocket.eventEmitter.on("message", _onClientChatMessage);
```

Comment:
The naming here isn't very consistent. I'd prefer to see the private functions also start with `_onChatClient`, because starting with `_onClient` made me confuse it with the Discord client for a moment.